### PR TITLE
[UTOPIA-1249] - (frontend) update UI to add questions

### PR DESCRIPTION
**Pull request by NoorChasib**
# Description

This PR includes the following proposed change(s):

- [UTOPIA-1249](https://apps.itsm.gov.bc.ca/jira/browse/UTOPIA-1249)
- create new UI components for related enactment and operational PIAs
- add ability to add and delete related enactment and operational PIAs
- add Initiative Summary component with character count
- add code comments for improved readability

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Updating Testing Framework(s)
- [ ] Version change

## How Has This Been Tested?

locally tested

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readable 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [x] I have performed a self-review of my own code
- [x] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [ ] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

File: src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx [hunk 1]
```diff
@@ -33,21 +44,84 @@ const PPQ = ({ printPreview }: IPPQProps) => {
...
+  // Manage form state using React's useState hook
   const [ppqForm, setPpqForm] = useState<IPPQ>(initialFormState);
 
+  // Function to handle state changes for ppqForm
   const stateChangeHandler = (value: any, key: keyof IPPQ) => {
     setNestedReactState(setPpqForm, key, value);
   };
 
+  // State for character count
+  const [charCount, setCharCount] = useState<number>(0);
```
Comment (BradyMitch) [inline src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx:69]: Should maybe name something like pidSummaryCharCount as its specific to pid summary.

File: src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx [hunk 2]
```diff
@@ -33,21 +44,84 @@ const PPQ = ({ printPreview }: IPPQProps) => {
...
+
+  // Max character count for initiative summary
+  const MAX_CHAR_COUNT = 500;
```
Comment (BradyMitch) [inline src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx:72]: Same with the name here.

File: src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx [hunk 3]
```diff
@@ -33,21 +44,84 @@ const PPQ = ({ printPreview }: IPPQProps) => {
...
+  };
+
+  // Handle character count in Initiative Summary
+  const handleInitiativeSummaryChange = (value = '') => {
+    stateChangeHandler(value, 'pidInitiativeSummary');
+    setCharCount(value.length);
+  };
+
+  // Display remaining character count or maximum limit
+  const getCharDisplayMessage = () =>
```
Comment (BradyMitch) [inline src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx:119]: Same here. Maybe getPidSummaryCharDisplayMessage

File: src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx (hunk 1 above)
Comment (NoorChasib) [reply src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx]: updated

File: src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx (hunk 2 above)
Comment (NoorChasib) [reply src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx]: updated

File: src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx (hunk 3 above)
Comment (NoorChasib) [reply src/frontend/src/components/public/PIAFormTabs/ppq/index.tsx]: updated

**Review by BradyMitch — APPROVED** [review]
Great job!

**Pull request merged**